moviebot/nlu/annotation/joint_bert/dataset.py

```diff
@@ -0,0 +1,187 @@
+"""Dataset loading for training and evaluating the JointBERT model. """
+import os
+import re
+from typing import Dict, Generator, List, Tuple
+
+import torch
+import yaml
+from torch.utils.data import Dataset
+from transformers import BertTokenizer
+
+from moviebot.nlu.annotation.joint_bert.slot_mapping import (
+    JointBERTIntent,
+    JointBERTSlot,
+)
+
+DataPoint = Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]
+
+_IGNORE_INDEX = -100
+_TOKENIZER_PATH = "bert-base-uncased"
+
+
+def load_yaml(path: str) -> Dict[str, List[str]]:
+    """Loads the YAML file at the given path.
+
+    Args:
+        path: The path to the YAML file.
+
+    Returns:
+        The data in the YAML file.
+    """
+    if not os.path.isfile(path):
+        raise FileNotFoundError(f"File not found: {path}")
+
+    with open(path) as f:
+        return yaml.safe_load(f)
+
+
+def parse_data(data: Dict[str, List[str]]) -> Generator[Tuple, None, None]:
```

Comment:
Nit: not sure the return type is correct here.

Reply:
It should be correct, but I could make Tuple even more explicit. None in `Generator[Tuple, None, None]` refers to SendType and ReturnType. First is the type of parameter the generator can recieve, while the second is the return type generetor returns when done iterating.